File: labs/03/old/year_codes.cpp
/* year_codes.cpp is a driver for function yearCode().
 * Begun by: Dr. Adams, for CS 214 at Calvin College.
 ****************************************************************/

#include <iostream>                      // interactive I/O
#include <string>                        // string class
using namespace std;

// REPLACE THIS LINE WITH THE DECLARATION OF yearCode()

int main() {
  cout << "\nEnter your academic year: "; // prompt
  string year;
  cin >> year;                            // read year
  cout << yearCode(year) << endl;         // display its code
}


/***************************************************************
 * yearCode() converts an academic year into its integer code. *
 * Receive: year, a string.
 * PRE: year is one of {freshman, sophomore, junior, or senior}.
 * Return: the integer code corresponding to year.
 */

// REPLACE THIS LINE WITH THE DEFINITION OF yearCode()

File: labs/03/old/yearCodes.f90
! yearCode.f90 computes the integer code for a given academic year.
! Dr. Adams, for CS 214 at Calvin College.
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

PROGRAM YearCodes

   IMPLICIT NONE                         ! disallow undeclared vars
   CHARACTER(12) :: year                 ! declare string var
                                         ! declare function

   PRINT*, 'Enter your academic year: '  ! prompt
   READ*, year                           ! get year
   PRINT*, YearCode(year)                ! call function & display

CONTAINS

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
! YearCode() computes the integer code for a given academic year.!
! Receive: year, a string.                                       !
! PRE: year is one of {freshman, sophomore, junior, senior}.     !
! Return: the code corresponding to year.                        !
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

! Replace this line with a definition of YearCode() here ... 
FUNCTION YearCode(year) RESULT code
  IMPLICIT NONE
  CHARACTER(12) :: year
  INTEGER :: code

END FUNCTION YearCode

END PROGRAM
